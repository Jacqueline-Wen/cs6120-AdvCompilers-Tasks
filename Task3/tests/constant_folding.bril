@main {
  a: int = const 5;
  b: int = const 8;
  c: bool = le a b;
  d: bool = not c;
  print d;
}