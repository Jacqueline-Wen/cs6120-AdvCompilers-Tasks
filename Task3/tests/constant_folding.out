{
    "functions": [
        {
            "instrs": [
                {
                    "dest": "d",
                    "op": "const",
                    "type": "bool",
                    "value": false
                },
                {
                    "args": [
                        "d"
                    ],
                    "op": "print"
                }
            ],
            "name": "main"
        }
    ]
}
